# Catalogues exploration
Ce notebook a pour but d'analyser et de rassembler les diffrentes sources de données afin de constituer un catalogue qui se veut être assez exhaustif accessible via une Web API.  

## 1. OpenNGC catalogue
 Ce catalogue a été télécharger depuis [ce lien](https://github.com/mattiaverga/OpenNGC). Il contient toutes les données présentes dans le catalogue NGC.

In [36]:
#import sys
import os
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 500)
CHEMIN_SOURCES = os.path.join("..","1_Data_brute")

In [2]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['NGC_IC_designation', 
    'object_type',
    'Right_Ascension',
    'Declinaison',
    'Constelation_name_abrv',
    'major_axis',
    'minor_axis',
    'major__axis_position_angle', 
    'B_Apparent_Magnitude',
    'V_Apparent_Magnitude',             
    'J_Apparent_Magnitude',
    'H_Apparent_Magnitude',
    'K_Apparent_Magnitude',
    'Mean_surface_brigthness', 
    'Hubble_morphological_type',
    'center_star_U_maghitude',
    'center_star_B_maghitude',
    'center_star_V_maghitude',
    'Messier_number',
    'NGC_numner',
    'IC_number',
    'Center_star_name',
    'Identifiers',
    'Common_names',       
    'NED_notes',  
    'OpenNGC_notes']

NGC_mattiaAverga_GitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "NGC_mattiaAverga_GitHub.csv"), sep=';',header=0, names=col_names)
NGC_mattiaAverga_GitHub.shape

(13958, 26)

Dans cette [source](http://messier.obspm.fr/m-q&a.html), on peut lire:
- M24: the Milky Way Patch in Sagittarius; it contains however the 11th magnitude cluster NGC 6603 which is sometimes erroneously listed as M24. Also, it may be that IC 4715 is M24. 
- M25: this is IC 4725, 
- M40: the double star Winnecke 4
- M45: the Pleiades; however, this cluster is associated with nebulae which have NGC numbers.

In [6]:
NGC_mattiaAverga_GitHub.loc[(NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4725') |
                        (NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4715')]

NGC_IC_designation object_type Right_Ascension  Declinaison  \
4923             IC4715        *Ass     18:16:56.12  -18:30:52.4   
4933             IC4725         OCl     18:31:46.77  -19:06:53.8   

     Constelation_name_abrv  major_axis  minor_axis  \
4923                    Sgr       120.0        60.0   
4933                    Sgr        14.1         NaN   

      major__axis_position_angle  B_Apparent_Magnitude  V_Apparent_Magnitude  \
4923                         NaN                   NaN                   4.5   
4933                         NaN                  5.29                   4.6   

      J_Apparent_Magnitude  H_Apparent_Magnitude  K_Apparent_Magnitude  \
4923                   NaN                   NaN                   NaN   
4933                   NaN                   NaN                   NaN   

      Mean_surface_brigthness Hubble_morphological_type  \
4923                      NaN                       NaN   
4933                      NaN                       NaN   

      center_star_U_maghitude  center_star_B_maghitude  \
4923                      NaN                      NaN   
4933                      NaN                      NaN   

      center_star_V_maghitude  Messier_number NGC_numner IC_number  \
4923                      NaN            24.0        NaN       NaN   
4933                      NaN            25.0        NaN       NaN   

     Center_star_name Identifiers          Common_names  \
4923              NaN         NaN  Small Sgr Star Cloud   
4933              NaN         NaN                   NaN   

                  NED_notes                             OpenNGC_notes  
4923  Milky Way star cloud.  V-mag taken from HEASARC's messier table  
4933                    NaN                                       NaN

*Note:* Les objets M24 et  M25 existent dans NGC_mattiaAverga_GitHub. Mais M40 et M45 n'y sont pas. Ils sont dans l'Addendum.

### Addendum
Le fichier Addendum contient des objets qui ne font partie des catalogue NGC ou IC mais qui peuvent interessé les amateurs de l'astronomie. Les objets M40 et M45 du catalogue de Messier n'ont pas d'identifiant NGC donc ils sont présents dans cet addendum (voir [source](https://github.com/mattiaverga/OpenNGC/blob/master/README.md)).

In [3]:
# Lecture de données depuis le fichier source vers un objet DataFrame
addendum_NGC_mattiaAverga_GitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "addendum_NGC_mattiaAverga_GitHub.csv"), sep=';', header=0, names=col_names)
addendum_NGC_mattiaAverga_GitHub.shape

(20, 26)

## 2. Construction du catalogue de Messier
Dans ce qui suit, nous allons construire notre catalogue de Messier à partir de plusieurs sources.

### 2.1. Le catalogue de Messier: *'Datastro'*
[Source](https://www.datastro.eu/explore/dataset/catalogue-de-messier/table/?disjunctive.objet&disjunctive.mag&disjunctive.english_name_nom_en_anglais&disjunctive.french_name_nom_francais&disjunctive.latin_name_nom_latin&sort=messier)

#### 2.1.1. Lecture du fichier

In [4]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Messier_number',
    'NGC_IC_designation', 
    'object_type',
    'Season',
    'Magnitude',
    'Constellation_EN',
    'Constellation_FR',
    'Constellation_Latin',
    'Right_Ascension',        
    'Declinaison',
    'Distance_light_year',
    'Size',
    'Discoverer',
    'Year',
    'Image1',
    'Image2',
    'Constelation_abr' ]
messier_Datastro_website = pd.read_csv(os.path.join(CHEMIN_SOURCES, "messier_Datastro_website.csv"), sep=';', header=0, names=col_names)
messier_Datastro_website.shape

(110, 17)

In [5]:
# Corrections d'un identifiant NGC 
messier_Datastro_website.loc[messier_Datastro_website['NGC_IC_designation']=='NGC 9176', 'NGC_IC_designation'] = 'NGC 1976'

#### 2.1.2. Division du catalogue de Messier

Nous avons besoin de traiter à part les deux objets M40 et M45. Car ces deux objets vont être mergés avec les données présentes dans le *Addendum* et le reste des objets Messier vont être mergés avec les données du *NGC_mattiaAverga_GitHub*.

In [7]:
messier_Datastro_website_without_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']!='M40') &
                        (messier_Datastro_website['Messier_number']!='M45')].copy()

messier_Datastro_website_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']=='M40') |
                        (messier_Datastro_website['Messier_number']=='M45')].copy()

#### 2.1.3. Merging entre Messier et NGC

In [38]:
# Normalisation du NGC_designation: NGCnnnn (suppression des vides)
messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==8
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "")

messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==7
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "0")

In [39]:
# Merge du dataframe qui contient tout les objets sauf M40 et M45
merge_messier_Datastro_website_without_M40_M45 = messier_Datastro_website_without_M40_M45.merge(NGC_mattiaAverga_GitHub, how='left', on='NGC_IC_designation')
merge_messier_Datastro_website_without_M40_M45.shape

(108, 42)

*Liste des features:*

- Messier_number_x
- NGC_IC_designation
- *object_type_x* (delete object_type_y is abreviation) 
- Season                         
- Magnitude                       
- Constellation_EN               
- Constellation_FR               
- Constellation_Latin            
- Right_Ascension_x  1 Nan(keep Right_Ascension_y no nan)            
- Declinaison_x  1 Nan(keep Declinaison_y no nan)                
- Distance_light_year           
- Size                           
- Discoverer                     
- Year                          
- Image1                         
- Image2                         
- *Constelation_abr (Delete)*             
- *object_type_y  (delete)*            
- *Right_Ascension_y*             
- *Declinaison_y*                  
- Constelation_name_abrv      
- major_axis                   
- minor_axis                    
- major__axis_position_angle    
- B_Apparent_Magnitude          
- V_Apparent_Magnitude          
- J_Apparent_Magnitude          
- H_Apparent_Magnitude          
- K_Apparent_Magnitude          
- Mean_surface_brigthness       
- Hubble_morphological_type      
- center_star_U_maghitude       
- center_star_B_maghitude       
- center_star_V_maghitude       
- *Messier_number_y (delete)*             
- NGC_numner  (6 double-listed in NGC)                  
- *IC_number (delete: 108 Nan)*                    
- Center_star_name               
- Identifiers                    
- Common_names                   
- NED_notes                      
- OpenNGC_notes

In [40]:
# Suppression des colunne inutiles ou redondantes
drop_list = ['object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'Messier_number_y', 'IC_number']
rename_lis = {'Messier_number_x' : 'Messier_number',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_without_M40_M45.shape

(108, 37)

In [11]:
# Extraction des objets M40 et M45 depuis l'Addendum
Open_NGC_M40_M45 = addendum_NGC_mattiaAverga_GitHub[~addendum_NGC_mattiaAverga_GitHub['Messier_number'].isna()].copy()
# Ajout du Numero Messier standardisé
Open_NGC_M40_M45['Messier_number_bis'] = 'M'+Open_NGC_M40_M45['Messier_number'].astype('int').astype('str')

# Remplacement de la colonne Messier_number par la nouvelle colonne
Open_NGC_M40_M45 = Open_NGC_M40_M45.drop(columns='Messier_number')
Open_NGC_M40_M45 = Open_NGC_M40_M45.rename(columns={'Messier_number_bis': 'Messier_number'})
Open_NGC_M40_M45

NGC_IC_designation object_type Right_Ascension Declinaison  \
13               M040          **      12:22:16.1   +58:05:04   
14             Mel022         OCl      03:47:28.6   +24:06:19   

   Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
13                    UMa         NaN         NaN                         NaN   
14                    Tau       150.0       150.0                        90.0   

    B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
13                   NaN                   8.0                   NaN   
14                   NaN                   1.2                   NaN   

    H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
13                   NaN                   NaN                      NaN   
14                   NaN                   NaN                      NaN   

   Hubble_morphological_type  center_star_U_maghitude  \
13                       NaN                      NaN   
14                       NaN                      NaN   

    center_star_B_maghitude  center_star_V_maghitude  NGC_numner  IC_number  \
13                      NaN                      NaN         NaN        NaN   
14                      NaN                      NaN         NaN        NaN   

    Center_star_name        Identifiers Common_names NED_notes  \
13               NaN  WDS J12222+5805AB          NaN       NaN   
14               NaN          MWSC 0305     Pleiades       NaN   

                                        OpenNGC_notes Messier_number  
13            VMag taken from HEASARC's messier table            M40  
14  Coordinates taken from Simbad, VMag taken from...            M45

In [12]:
# Merge du dataframe qui contient M40 et M45
merge_messier_Datastro_website_M40_M45 = messier_Datastro_website_M40_M45.merge(Open_NGC_M40_M45, how='left', on='Messier_number')
merge_messier_Datastro_website_M40_M45

Messier_number NGC_IC_designation_x                object_type_x  \
0            M45                  NaN   Open Cluster / Amas Ouvert   
1            M40           Winnecke 4  Double star / Étoile Double   

               Season  Magnitude Constellation_EN Constellation_FR  \
0      Winter / Hiver          1              NaN              NaN   
1  Spring / Printemps          9              NaN              NaN   

  Constellation_Latin Right_Ascension_x Declinaison_x  Distance_light_year  \
0                 NaN               NaN           NaN                410.0   
1                 NaN               NaN           NaN                  NaN   

     Size Discoverer    Year                                Image1  \
0  120,0'        NaN     NaN  http://www.lasam.ca/messier/M045.JPG   
1     NaN   Hevelius  1660.0  http://www.lasam.ca/messier/M040.JPG   

                                              Image2 Constelation_abr  \
0  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   
1  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   

  NGC_IC_designation_y object_type_y Right_Ascension_y Declinaison_y  \
0               Mel022           OCl        03:47:28.6     +24:06:19   
1                 M040            **        12:22:16.1     +58:05:04   

  Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
0                    Tau       150.0       150.0                        90.0   
1                    UMa         NaN         NaN                         NaN   

   B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
0                   NaN                   1.2                   NaN   
1                   NaN                   8.0                   NaN   

   H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
0                   NaN                   NaN                      NaN   
1                   NaN                   NaN                      NaN   

  Hubble_morphological_type  center_star_U_maghitude  center_star_B_maghitude  \
0                       NaN                      NaN                      NaN   
1                       NaN                      NaN                      NaN   

   center_star_V_maghitude  NGC_numner  IC_number  Center_star_name  \
0                      NaN         NaN        NaN               NaN   
1                      NaN         NaN        NaN               NaN   

         Identifiers Common_names NED_notes  \
0          MWSC 0305     Pleiades       NaN   
1  WDS J12222+5805AB          NaN       NaN   

                                       OpenNGC_notes  
0  Coordinates taken from Simbad, VMag taken from...  
1            VMag taken from HEASARC's messier table

In [13]:
# Move data before deleting NGC_IC_designation_y
merge_messier_Datastro_website_M40_M45.loc[merge_messier_Datastro_website_M40_M45['NGC_IC_designation_x'].isna(),
                                          'NGC_IC_designation_x'] = merge_messier_Datastro_website_M40_M45['NGC_IC_designation_y']

In [14]:
# Suppression des colunne inutiles ou redondantes
drop_list = ['NGC_IC_designation_y', 'object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'IC_number']
rename_lis = {'NGC_IC_designation_x': 'NGC_IC_designation',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_M40_M45.shape

(2, 37)

#### 2.1.4. Concatination entre Messier contenant les objets M40 et M45 et le Messier ayant le reste des objets

In [15]:
messier_catalogue = pd.concat([merge_messier_Datastro_website_without_M40_M45, merge_messier_Datastro_website_M40_M45])

In [16]:
messier_catalogue[messier_catalogue['Hubble_morphological_type'].isna()]

Messier_number NGC_IC_designation  \
2              M76            NGC0650   
3              M80            NGC6093   
4              M93            NGC2447   
7              M37            NGC2099   
8              M36            NGC1960   
9              M48            NGC2548   
14             M16            NGC6611   
15              M4            NGC6121   
16             M15            NGC7078   
17             M18            NGC6613   
18             M21            NGC6531   
19             M35            NGC2168   
21             M73            NGC6994   
23             M78            NGC2068   
26             M69            NGC6637   
27             M57            NGC6720   
29             M44            NGC2632   
31             M72            NGC6981   
32              M3            NGC5272   
33              M7            NGC6475   
34              M5            NGC5904   
35             M39            NGC7092   
36             M29            NGC6913   
37            M107            NGC6171   
39             M34            NGC1039   
40             M56            NGC6779   
43              M9            NGC6333   
44              M1            NGC1952   
45             M22            NGC6656   
46             M30            NGC7099   
..             ...                ...   
60             M68            NGC4590   
63             M47            NGC2422   
65             M79            NGC1904   
67             M20            NGC6514   
68             M10            NGC6254   
69             M14            NGC6402   
70             M24            NGC6603   
73             M75            NGC6864   
75             M41            NGC2287   
76             M46            NGC2437   
80              M8            NGC6523   
81             M11            NGC6705   
82             M23            NGC6494   
83             M25             IC4725   
84             M54            NGC6715   
85             M62            NGC6266   
87             M67            NGC2682   
88             M71            NGC6838   
91            M103            NGC0581   
93             M70            NGC6681   
94             M50            NGC2323   
97             M97            NGC3587   
99             M92            NGC6341   
100            M90            NGC4560   
102             M2            NGC7089   
104            M17            NGC6618   
105            M27            NGC6853   
106            M28            NGC6626   
0              M45             Mel022   
1              M40         Winnecke 4   

                                   object_type              Season  Magnitude  \
2      Planetary Nebula / Nébuleuse Planétaire    Autumn / Automne         10   
3           Globular Cluster / Amas Globulaire        Summer / Été          7   
4                   Open Cluster / Amas Ouvert      Winter / Hiver          6   
7                   Open Cluster / Amas Ouvert      Winter / Hiver          5   
8                   Open Cluster / Amas Ouvert      Winter / Hiver          6   
9                   Open Cluster / Amas Ouvert      Winter / Hiver          5   
14      Emission Nebula / Nébuleuse à émission        Summer / Été          6   
15          Globular Cluster / Amas Globulaire        Summer / Été          5   
16          Globular Cluster / Amas Globulaire    Autumn / Automne          6   
17                  Open Cluster / Amas Ouvert        Summer / Été          6   
18                  Open Cluster / Amas Ouvert        Summer / Été          5   
19                  Open Cluster / Amas Ouvert      Winter / Hiver          5   
21                  Open Cluster / Amas Ouvert    Autumn / Automne          9   
23   Reflection Nebula / Nébuleuse à réflexion      Winter / Hiver          8   
26          Globular Cluster / Amas Globulaire        Summer / Été          7   
27     Planetary Nebula / Nébuleuse Planétaire        Summer / Été          9   
29                  Open Cluster / Amas Ouvert  Spring / Printemps          3   
31 

### 2.3. Le catalogue de Messier: *'jbcurtin_gitHub'*
Ce catalogue ([Voir référence](https://github.com/jbcurtin/messier-catalogue)) est pricipalement intéressant pour les images qu'il contient pour chaque objet.

In [17]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Messier_number_jbcurtin_gitHub',
    'NGC_IC_designation_jbcurtin_gitHub', 
    'Common_name_jbcurtin_gitHub',
    'image_jbcurtin_gitHub',
    'Object_type_jbcurtin_gitHub',
    'Distance_KLY_jbcurtin_gitHub',             
    'Constellation_jbcurtin_gitHub',
    'Apparent_magnitude_jbcurtin_gitHub',
    'Right_Ascension_jbcurtin_gitHub',        
    'Declinaison_jbcurtin_gitHub']
messier_with_picture_links_jbcurtin_gitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "messier_with_picture_links_jbcurtin_gitHub.csv"), keep_default_na=True, sep=';', header=0, names=col_names)
messier_with_picture_links_jbcurtin_gitHub.loc[messier_with_picture_links_jbcurtin_gitHub['Common_name_jbcurtin_gitHub']==' ',
                       'Common_name_jbcurtin_gitHub'] = np.nan #not a space
    
messier_with_picture_links_jbcurtin_gitHub.dtypes

Messier_number_jbcurtin_gitHub         object
NGC_IC_designation_jbcurtin_gitHub     object
Common_name_jbcurtin_gitHub            object
image_jbcurtin_gitHub                  object
Object_type_jbcurtin_gitHub            object
Distance_KLY_jbcurtin_gitHub           object
Constellation_jbcurtin_gitHub          object
Apparent_magnitude_jbcurtin_gitHub    float64
Right_Ascension_jbcurtin_gitHub        object
Declinaison_jbcurtin_gitHub            object
dtype: object

In [18]:
# Merge entre 'messier_catalogue' et le nouveau dataset
messier_catalogue2 = messier_with_picture_links_jbcurtin_gitHub.merge(messier_catalogue,how='left', left_on='Messier_number_jbcurtin_gitHub', right_on='Messier_number')
messier_catalogue2.shape

(110, 47)

In [19]:
#keep 'Common_name_jbcurtin_gitHub', delete_Common_names
messier_catalogue2.loc[messier_catalogue2['Common_name_jbcurtin_gitHub'].isna(),
                   'Common_name_jbcurtin_gitHub'] = messier_catalogue2['Common_names']

# In 'Object_type_jbcurtin_gitHub', Galaxy is subtyped. So we should keep that info
# Keep 'object_type', delete 'Object_type_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['object_type'].str.contains('Galaxy'),
                   'object_type'] = messier_catalogue2['Object_type_jbcurtin_gitHub']

#Delete 'Constellation_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['Constellation_Latin'].isna(),
                   'Constellation_Latin'] = messier_catalogue2['Constellation_jbcurtin_gitHub']

drop_list = ['Common_names', 'Messier_number_jbcurtin_gitHub', 'NGC_IC_designation_jbcurtin_gitHub', 
             'Object_type_jbcurtin_gitHub', 'Distance_KLY_jbcurtin_gitHub', 'Constellation_jbcurtin_gitHub',
            'Right_Ascension_jbcurtin_gitHub', 'Declinaison_jbcurtin_gitHub']
rename_lis = {'Common_name_jbcurtin_gitHub': 'Common_name'}

messier_catalogue2 = messier_catalogue2.drop(columns=drop_list)
messier_catalogue2 = messier_catalogue2.rename(columns=rename_lis)
messier_catalogue2.shape


(110, 39)

### 2.4. Le catalogue de Messier: *'Nexstarsite'*

https://www.nexstarsite.com/Book/DSO.htm

In [20]:
MessierObjects_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite", "MessierObjects_nexstarsite.xls"), skiprows=2)
MessierObjects_nexstarsite.dtypes

ObjectNum          int64
Name              object
Type              object
Constellation     object
RAHour             int64
RAMinute         float64
DecSign           object
DecDeg             int64
DecMinute        float64
Magnitude         object
Info              object
Distance (ly)      int64
dtype: object

Ce dataset ne contient pas de nouvelles données.

In [21]:
messier_catalogue_final = messier_catalogue2

In [22]:
messier_catalogue_final.dtypes

Common_name                            object
image_jbcurtin_gitHub                  object
Apparent_magnitude_jbcurtin_gitHub    float64
Messier_number                         object
NGC_IC_designation                     object
object_type                            object
Season                                 object
Magnitude                               int64
Constellation_EN                       object
Constellation_FR                       object
Constellation_Latin                    object
Distance_light_year                   float64
Size                                   object
Discoverer                             object
Year                                  float64
Image1                                 object
Image2                                 object
Constelation_abr                       object
Right_Ascension                        object
Declinaison                            object
Constelation_name_abrv                 object
major_axis                        

## 3. Construction du catalogue de Caldwell

In [161]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Caldwell_number',
    'NGC_IC_designation', 
    'object_type',
    'Constelation_abr',
    'Magnitude',
    'Size',
    'Right_Ascension_Hour',
    'Right_Ascension_Minute', 
    'Declinaison_Sign',
    'Declinaison_Deg',
    'Declinaison_Minute'
     ]

CaldwellObjects_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite", "CaldwellObjects_nexstarsite.xls"), skiprows=2, keep_default_na=True, names=col_names)
CaldwellObjects_nexstarsite.shape

(109, 11)

In [162]:
# Normalisation du NGC_designation: NGCnnnn (suppression des vides)
CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==8) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("NGC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "")

CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==7) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("NGC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "0")

CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==6) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("NGC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "00")

In [163]:
# Normalisation du NGC_designation: ICnnnn (suppression des vides)
CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==7) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("IC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "")

CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==6) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("IC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "0")

CaldwellObjects_nexstarsite.loc[( (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==5) &
                                (CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("IC")) ), 
                                'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "00")



In [165]:
CaldwellObjects_nexstarsite[~(CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("IC")| CaldwellObjects_nexstarsite['NGC_IC_designation'].str.contains("NGC") )]

Caldwell_number NGC_IC_designation    object_type Constelation_abr  \
8                 9            Sh2-155  Bright Nebula              Cep   
40               41               none   Open Cluster              Tau   
98               99               none    Dark Nebula              Cru   

   Magnitude             Size  Right_Ascension_Hour  Right_Ascension_Minute  \
8         --    Size: 50 x 10                    22                    56.8   
40       0.5        Size: 330                     4                    27.0   
98        --  Size: 400 x 300                    12                    53.0   

   Declinaison_Sign  Declinaison_Deg  Declinaison_Minute  
8                 +               62                  37  
40                +               16                   0  
98                -               63                   0

The objects C9, C41 and C99 have no NGC/IC designation

In [172]:
# Merge entre 'messier_catalogue' et le nouveau dataset
cadlwell_catalogue = CaldwellObjects_nexstarsite.merge(NGC_mattiaAverga_GitHub, how='left', on='NGC_IC_designation')
cadlwell_catalogue.shape

(109, 36)

Liste des colonnes:

- Caldwell_number   (Keep)      
- NGC_IC_designation  (Keep)         
- object_type_x    (keep -rename)             
- Constelation_abr  (keep -rename)           
- Magnitude   (delete)            
- Size   (Keep)                        
- Right_Ascension_Hour  (delete)
- Right_Ascension_Minute  (delete)
- Declinaison_Sign  (delete)
- Declinaison_Deg  (delete)
- Declinaison_Minute (delete)
- object_type_y  (delete)
- Right_Ascension (Keep) 
- Declinaison (Keep) 
- Constelation_name_abrv (delete)
- major_axis (Keep) 
- minor_axis (Keep) 
- major__axis_position_angle (Keep) 
- B_Apparent_Magnitude  (Keep) 
- V_Apparent_Magnitude  (Keep) 
- J_Apparent_Magnitude (Keep) 
- H_Apparent_Magnitude  (Keep) 
- K_Apparent_Magnitude  (Keep) 
- Mean_surface_brigthness  (Keep) 
- Hubble_morphological_type  (Keep) 
- center_star_U_maghitude   (Keep) 
- center_star_B_maghitude (Keep) 
- center_star_V_maghitude   (Keep) 
- Messier_number  (delete)
- NGC_numner     (Keep) 
- IC_number     (Keep) 
- Center_star_name (Keep) 
- Identifiers     (Keep) 
- Common_names     (Keep) 
- NED_notes       (Keep) 
- OpenNGC_notes   (Keep) 

In [173]:
drop_list = ["Magnitude", "Right_Ascension_Hour", "Right_Ascension_Minute", "Declinaison_Sign", 
            "Declinaison_Deg", "Declinaison_Minute", "object_type_y", "Constelation_name_abrv", "Messier_number"]
rename_lis = {'object_type_x': 'object_type',
             'Constelation_abr': 'Constelation_name_abrv',}

cadlwell_catalogue = cadlwell_catalogue.drop(columns=drop_list)
cadlwell_catalogue = cadlwell_catalogue.rename(columns=rename_lis)

In [174]:
# Ajout du Numero Messier standardisé
cadlwell_catalogue['Caldwell_number_bis'] = 'S'+cadlwell_catalogue['Caldwell_number'].astype('int').astype('str')

# Remplacement de la colonne Messier_number par la nouvelle colonne
cadlwell_catalogue = cadlwell_catalogue.drop(columns='Caldwell_number')
cadlwell_catalogue = cadlwell_catalogue.rename(columns={'Caldwell_number_bis': 'Caldwell_number'})
cadlwell_catalogue

NGC_IC_designation        object_type Constelation_name_abrv  \
0              NGC0188       Open Cluster                    Cep   
1              NGC0040   Planetary Nebula                    Cep   
2              NGC4236      Spiral Galaxy                    Dra   
3              NGC7023      Bright Nebula                    Cep   
4               IC0342      Spiral Galaxy                    Cam   
5              NGC6543   Planetary Nebula                    Dra   
6              NGC2403      Spiral Galaxy                    Cam   
7              NGC0559       Open Cluster                    Cas   
8              Sh2-155      Bright Nebula                    Cep   
9              NGC0663       Open Cluster                    Cas   
10             NGC7635      Bright Nebula                    Cas   
11             NGC6946      Spiral Galaxy                    Cep   
12             NGC0457       Open Cluster                    Cas   
13         NGC 869/884       Open Cluster                    Per   
14             NGC6826   Planetary Nebula                    Cyg   
15             NGC7243       Open Cluster                    Lac   
16             NGC0147  Elliptical Galaxy                    Cas   
17             NGC0185  Elliptical Galaxy                    Cas   
18              IC5146      Bright Nebula                    Cyg   
19             NGC7000      Bright Nebula                    Cyg   
20             NGC4449   Irregular Galaxy                    CVn   
21             NGC7662   Planetary Nebula                    And   
22             NGC0891      Spiral Galaxy                    And   
23             NGC1275   Irregular Galaxy                    Per   
24             NGC2419   Globular Cluster                    Lyn   
25             NGC4244      Spiral Galaxy                    CVn   
26             NGC6888      Bright Nebula                    Cyg   
27             NGC0752       Open Cluster                    And   
28             NGC5005      Spiral Galaxy                    CVn   
29             NGC7331      Spiral Galaxy                    Peg   
..                 ...                ...                    ...   
79             NGC5139   Globular Cluster                    Cen   
80             NGC6352   Globular Cluster                    Ara   
81             NGC6193       Open Cluster                    Ara   
82             NGC4945      Spiral Galaxy                    Cen   
83             NGC5286   Globular Cluster                    Cen   
84              IC2391       Open Cluster                    Vel   
85             NGC6397   Globular Cluster                    Ara   
86             NGC1261   Globular Cluster                    Hor   
87             NGC5823       Open Cluster                    Cir   
88             NGC6087       Open Cluster                    Nor   
89             NGC2867   Planetary Nebula                    Car   
90             NGC3532       Open Cluster                    Car   
91             NGC3372      Bright Nebula                    Car   
92             NGC6752   Globular Cluster                    Pav   
93             NGC4755       Open Cluster                    Cru   
94             NGC6025       Open Cluster                    TrA   
95             NGC2516       Open Cluster                    Car   
96             NGC3766       Open Cluster                    Cen   
97             NGC4609       Open Cluster                    Cru   
98                none        Dark Nebula                    Cru   
99              IC2944       Open Cluster                    Cen   
100            NGC6744      Spiral Galaxy                    Pav   
101             IC2602       Open Cluster                    Car   
102            NGC2070      Bright Nebula                    Dor   
103            NGC0362   Globular Cluster                    Tuc   
104            NGC4833   Globular Cluster                    Mus   
105            NGC0104   Globular Cluster                   

In [175]:
caldwell_catalogue_final = cadlwell_catalogue

## 4. Construction du catalogue de Herschel400

https://www.go-astronomy.com/herschel-objects.htm

The Herschel 400 catalog is a subset of the original Herschel (2,500 object) catalog that contains 400 northern-sky deep-sky objects (galaxies, nebulae, and star clusters) that can be found by amateur astronomers that want a challenge after finding the easier/brighter Messier and Caldwell catalog objects. The Herschel 400 requires a bit larger telescope (6 inches and larger) than the Messier objects that can all be found in 4-inch scopes.

Herschel 400 objects are often designated by their NGC number in constellation charts rather than by a Herschel H designation, unlike Messier objects that are often designated by their M number..

In [193]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['NGC_IC_designation', 
    'name',
    'object_type',
    'Constelation_abr',
    'Right_Ascension_Hour',
    'Right_Ascension_Minute', 
    'Declinaison_Sign',
    'Declinaison_Deg',
    'Declinaison_Minute',
    'Magnitude',
    'info']

Herschel400_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite","Herschel400_nexstarsite.xls"), skiprows=2, keep_default_na=True, names=col_names)
Herschel400_nexstarsite

NGC_IC_designation             name         object_type Constelation_abr  \
0                    40       PK 120+9.1    Planetary Nebula              Cep   
1                   129          OCL 294        Open Cluster              Cas   
2                   136          OCL 295        Open Cluster              Cas   
3                   157       MCG 2-2-56              Galaxy              Cet   
4                   185          UGC 396              Galaxy              Cas   
5                   205            M 110              Galaxy              And   
6                   225          OCL 305        Open Cluster              Cas   
7                   246      PK 118-74.1    Planetary Nebula              Cet   
8                   247       ESO 540-22              Galaxy              Cet   
9                   253       ESO 474-29              Galaxy              Scl   
10                  278          UGC 528              Galaxy              Cas   
11                  288     ESO 474-SC37    Globular Cluster              Scl   
12                  381          OCL 317        Open Cluster              Cas   
13                  404          UGC 718              Galaxy              And   
14                  436          OCL 320        Open Cluster              Cas   
15                  457          OCL 321        Open Cluster              Cas   
16                  488          UGC 907              Galaxy              Psc   
17                  524          UGC 968              Galaxy              Psc   
18                  559          OCL 322        Open Cluster              Cas   
19                  584  IC 1712;H I 100              Galaxy              Cet   
20                  596        MCG 1-5-5              Galaxy              Cet   
21                  598             M 33              Galaxy              Tri   
22                  613       ESO 413-11              Galaxy              Scl   
23                  615        MCG 1-5-8              Galaxy              Cet   
24                  637          OCL 329        Open Cluster              Cas   
25                  651             M 76    Planetary Nebula              Per   
26                  654          OCL 330        Open Cluster              Cas   
27                  659          OCL 332        Open Cluster              Cas   
28                  663          OCL 333        Open Cluster              Cas   
29                  720       MCG 2-5-68              Galaxy              Cet   
..                  ...              ...                 ...              ...   
370                6939          OCL 217        Open Cluster              Cep   
371                6940          OCL 141        Open Cluster              Vul   
372                6946        UGC 11597              Galaxy              Cep   
373                7000          LBN 373              Nebula              Cyg   
374                7006          GCL 119    Globular Cluster              Del   
375                7008        PK 93+5.2    Planetary Nebula              Cyg   
376                7009       PK 37-34.1    Planetary Nebula              Aqr   
377                7044          OCL 198        Open Cluster              Cyg   
378                7062          OCL 205        Open Cluster              Cyg   
379                7086          OCL 214        Open Cluster              Cyg   
380                7128          OCL 218        Open Cluster              Cyg   
381                7142          OCL 241        Open Cluster              Cep   
382                7160          OCL 236        Open Cluster              Cep   
383                7209          OCL 215        Open Cluster              Lac   
384                7217        UGC 11914              Galaxy              Peg   
385                7243          OCL 221        Open Cluster              Lac   
386                7296          OCL 228        Open Cluster              Lac   
387                7331        UGC 1

In [192]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['NGC_IC_designation', 
    'name',
    'object_type',
    'Constelation_abr',
    'Right_Ascension_Hour',
    'Right_Ascension_Minute', 
    'Declinaison_Sign',
    'Declinaison_Deg',
    'Declinaison_Minute',
    'Magnitude',
    'info']

Herschel400_astr_tom = pd.read_excel(os.path.join(CHEMIN_SOURCES,"Herschel_astro_tom.xls"), keep_default_na=True)
Herschel400_astr_tom

NGC  CON   TYPE  R.A.(2000)  Unnamed: 4  Dec  Unnamed: 6   Mag  \
0     205  AND     E6           0        40.4   41          41   8.0   
1     404  AND     E0           1         9.5   35          43  10.1   
2     752  AND  OPNCL           1        57.8   37          41   5.7   
3     891  AND     Sb           2        22.6   42          21  11.5   
4    7662  AND  PLNNB          23        25.9   42          33   8.6   
5    7686  AND  OPNCL          23        30.2   49           8   5.6   
6    6755  AQL  OPNCL          19         7.8    4          14   7.5   
7    6756  AQL  OPNCL          19         8.7    4          41  10.6   
8    6781  AQL  PLNNB          19        18.5    6          32  11.8   
9    7009  AQR  PLNNB          21         4.2  -11          22   8.3   
10   7606  AQR  GALXY          23        19.1   -8          29  11.5   
11   7723  AQR  GALXY          23        38.9  -12          58  12.0   
12   7727  AQR  GALXY          23        39.9  -12          18  11.5   
13    772  ARI     Sb           1        59.4   19           0  11.5   
14   1664  AUR  OPNCL           4        51.1   43          42   7.6   
15   1857  AUR  OPNCL           5        20.2   39          21   7.0   
16   1907  AUR  OPNCL           5        28.0   35          19   8.2   
17   1931  AUR  CL+NB           5        31.4   34          15  11.3   
18   2126  AUR  OPNCL           6         3.0   49          54  10.2   
19   2281  AUR  OPNCL           6        49.3   41           4   5.4   
20   5248  BOO     Sc          13        37.4    8          53  11.4   
21   5466  BOO  GLOCL          14         5.5   28          32   9.1   
22   5557  BOO     E1          14        18.3   36          29  12.2   
23   5676  BOO     Sc          14        32.8   49          27  11.7   
24   5689  BOO    SBa          14        35.5   48          44  12.7   
25   1501  CAM  PLNNB           4         7.0   60          55  12.0   
26   1502  CAM  OPNCL           4         7.7   62          20   5.7   
27   1961  CAM    SBp           5        42.2   69          23  11.5   
28   2403  CAM     Sc           7        36.8   65          37   9.5   
29   2655  CAM  S(B)a           8        55.6   78          13  11.5   
..    ...  ...    ...         ...         ...  ...         ...   ...   
370  4660  VIR     E5          12        44.6   11          12  12.1   
371  4665  VIR  GALXY          12        45.2    3           4  12.4   
372  4666  VIR     Sc          12        45.2    0          27  12.0   
373  4697  VIR  GALXY          12        48.6   -5          48  10.2   
374  4698  VIR  GALXY          12        48.5    8          30  12.1   
375  4699  VIR  GALXY          12        49.1   -8          40  10.5   
376  4753  VIR  GALXY          12        52.4   -1          12  11.7   
377  4754  VIR     SO          12        52.4   11          19  12.0   
378  4762  VIR     SO          12        53.0   11          14  11.1   
379  4781  VIR    SBc          12        54.4  -10          32  11.8   
380  4845  VIR     Sa          12        58.1    1          35  12.9   
381  4856  VIR  GALXY          12        59.3  -15           2  11.4   
382  4866  VIR     SO          12        59.4   14          10  11.9   
383  4900  VIR  GALXY          13         0.7    2          30  12.8   
384  4958  VIR     E6          13         5.7   -8           1  11.4   
385  4995  VIR     Sb          13         9.6   -7          50  11.9   
386  5054  VIR     Sb          13        16.9  -16          39  11.5   
387  5363  VIR     Ep          13        56.2    5          16  11.4   
388  5364  VIR   Sb+p          13        56.3    5           2  13.2   
389  5566  VIR    Sb+          14        20.4    3          56  12.0   
390  5576  VIR     E2          14        21.1    3          16  12.3   
391  5634  VIR  GLOCL          14        29.6   -5          59  11.0   
392  5746  VIR     Sb          14        45.0    1          49  12.3   
393  5846  VIR     E0          15         6.5 

In [187]:
NGC_mattiaAverga_GitHub[NGC_mattiaAverga_GitHub["NGC_IC_designation"]=="NGC0205"]

NGC_IC_designation object_type Right_Ascension  Declinaison  \
5829            NGC0205           G     00:40:22.08  +41:41:07.1   

     Constelation_name_abrv  major_axis  minor_axis  \
5829                    And       16.22        9.59   

      major__axis_position_angle  B_Apparent_Magnitude  V_Apparent_Magnitude  \
5829                       170.0                  8.92                  8.07   

      J_Apparent_Magnitude  H_Apparent_Magnitude  K_Apparent_Magnitude  \
5829                  6.45                  5.87                  5.59   

      Mean_surface_brigthness Hubble_morphological_type  \
5829                    23.78                         E   

      center_star_U_maghitude  center_star_B_maghitude  \
5829                      NaN                      NaN   

      center_star_V_maghitude  Messier_number NGC_numner IC_number  \
5829                      NaN           110.0        NaN       NaN   

     Center_star_name                                        Identifiers  \
5829              NaN  2MASX J00402207+4141070,IRAS 00376+4124,MCG +0...   

     Common_names NED_notes OpenNGC_notes  
5829          NaN       NaN           NaN

# Catalogues Merging

## Final Catalogue file

In [29]:
reorder = ['Messier_number', 'NGC_IC_designation', 'object_type', 'Common_name', 'Constellation_FR', 'Constellation_EN',
'Constellation_Latin', 'Constelation_abr','Constelation_name_abrv','Hubble_morphological_type', 'Discoverer', 'Year', 
'Season', 'Right_Ascension','Declinaison', 'Distance_light_year', 'Size', 'major_axis', 'minor_axis', 'major__axis_position_angle',
'Apparent_magnitude_jbcurtin_gitHub', 'Magnitude', 'B_Apparent_Magnitude', 'V_Apparent_Magnitude', 'J_Apparent_Magnitude',
'H_Apparent_Magnitude', 'K_Apparent_Magnitude', 'Mean_surface_brigthness',
'Center_star_name', 'center_star_U_maghitude', 'center_star_B_maghitude', 'center_star_V_maghitude',
'Image1', 'Image2', 'image_jbcurtin_gitHub', 'NGC_numner', 'Identifiers', 'NED_notes', 'OpenNGC_notes']

messier_catalogue_final = messier_catalogue_final[reorder]

In [176]:
#messier_catalogue_final.to_csv(os.path.join(CHEMIN_OUTPUT, "Final_catalogue.csv"),";")
messier_catalogue_final.to_csv("Final_messier_catalogue.csv",";")
caldwell_catalogue_final.to_csv("Final_caldwell_catalogue.csv",";")